src/test/java/spoon/test/model/SwitchCaseTest.java

```diff
@@ -365,4 +367,52 @@ public void test_addCaseAt_throwsIndexOutOfBoundsException_whenPositionIsOutOfBo
 			assertThrows(IndexOutOfBoundsException.class, () -> switchExpression.addCaseAt(3, onlyCase));
 		}
 	}
+	
+	@DisplayName("should return the first case expression")
+	@Test
+	public void test_getCaseExpression_shouldReturnTheFirstCaseExpression() {
+		Factory factory = new Launcher().getFactory();
+
+		CtCase<Integer> ctCase = factory.createCase();
+		CtExpression<Integer> first = factory.createCodeSnippetExpression("1");
+		CtExpression<Integer> second = factory.createCodeSnippetExpression("2");
+		CtExpression<Integer> third = factory.createCodeSnippetExpression("3");
+
+		ctCase.setCaseExpressions(Arrays.asList(first, second, third));
+
+		assertEquals(first, ctCase.getCaseExpression());
+	}
+
+	@Nested
+	class SetCaseExpressionInSwitchBeforeJava12 {
+		@DisplayName("should remove all existing case expressions and insert the specified case expression")
+		@Test
+		public void test_setCaseExpression_removeExistingCaseExpressionsAndInsertTheSpecified() {
+			Factory factory = new Launcher().getFactory();
+
+			CtCase<Integer> ctCase = factory.createCase();
+			CtExpression<Integer> first = factory.createCodeSnippetExpression("1");
+			CtExpression<Integer> second = factory.createCodeSnippetExpression("2");
+			CtExpression<Integer> third = factory.createCodeSnippetExpression("3");
+			CtExpression<Integer> fourth = factory.createCodeSnippetExpression("4");
+			ctCase.setCaseExpressions(Arrays.asList(first, second, third));
+
+			ctCase.setCaseExpression(fourth);
+
+			assertEquals(1, ctCase.getCaseExpressions().size());
+			assertEquals(fourth, ctCase.getCaseExpression());
+		}
+
+		@DisplayName("should clear the list of expressions when null is passed")
+		@Test
+		public void test_setCaseExpression_removeAllCaseExpressions() {
+			String code = "class A { public void f(int i) { switch(i) { case 1,2,3: break; } } }";
+			CtModel model = createModelFromString(code);
+			CtCase<?> ctCase = model.getElements(new TypeFilter<>(CtCase.class)).get(0);
+
+			ctCase.setCaseExpression(null);
+
+			assertTrue(ctCase.getCaseExpressions() instanceof EmptyClearableList);
```

Comment:
Like we've discussed a few times before, I would if at all possible avoid using `assertTrue` for anything but asserting something that is inherently a boolean. What you want to assert here isn't a boolean itself, but rather you want to check that the expression list is empty. I would also argue that asserting specifically against `EmptyClearableList` is overspecifying the test: that's an implementation detail that's not visible through the API.

I would change to this:

```suggestion
			assertThat(ctCase.getCaseExpressions().size(), equalTo(0));
```

As we just have the core Hamcrest matchers. But if we add all Hamcrest matchers as of version 2.2 (I'll open an issue about that), this assertion could be even more clear.

```suggestion
			assertThat(ctCase.getCaseExpressions(), is(empty()));
```